source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -45,42 +45,50 @@ public MessageExchangeStream(Stream stream, ILog log)
         public void IdentifyAsClient()
         {
             log.Write(EventType.Diagnostic, "Identifying as a client");
-            streamWriter.Write($"{MxClient} {currentVersion}");
-            streamWriter.WriteLine();
+            SendIdentityMessage($"{MxClient} {currentVersion}");
+            ExpectServerIdentity();
+        }
+
+        void SendControlMessage(string message)
+        {
+            streamWriter.WriteLine(message);
+            streamWriter.Flush();
+        }
+
+        async Task SendControlMessageAsync(string message)
+        {
+            await streamWriter.WriteLineAsync(message);
+            await streamWriter.FlushAsync();
+        }
+
+        void SendIdentityMessage(string identityLine)
```

Comment:
Ok, so I'm inferring from this that it's important (for some reason) for "identity" messages (but not "control" messages) to be followed by an extra newline. Have I got that right?

Reply:
Yep! I don't know why, and I haven't looked into it, but it's been that way for 6 years, so I've left it as-is. One thought that came to mind is that it might be for some conceptual similarity to HTTP, or left as a possible extension point (.e.g everything before a blank line is connection/identity metadata).